time-y1-nozzle.sh

```
@@ -1,81 +1,205 @@
-+    my_rhs(t, state)
-+
-+    #(current_step, current_t, current_state) = \
-+    #    advance_state(rhs=my_rhs, timestepper=timestepper,
-+    #                  checkpoint=my_checkpoint,+    #                  get_timestep=get_timestep, state=current_state,
-+    #                  t_final=t_final, t=current_t, istep=current_step,
-+    #                  logmgr=logmgr,eos=eos,dim=dim)
-
-     if rank == 0:
-         logger.info("Checkpointing final state ...")
-
-+    elapsed = time()
-+    with open("case-time.txt", "w") as outf:
-+        outf.write(f"{repr(time())} {repr(elapsed)}\n")
-+
-     my_checkpoint(current_step, t=current_t,
-                   dt=(current_t - checkpoint_t),
-                   state=current_state)
-EOF
-
-(cd CEESD-Y1_nozzle; patch -p1 < ../add-timing.patch)
-
 cd CEESD-Y1_nozzle/startup
-PYOPENCL_TEST=port:pthread python -m mpi4py nozzle.py
+DRIVER_HASH=$(git rev-parse main)
+
+# --- DEVELOPERS NOTE:
+# --- The following (sed) edit is fragile in that, like a patch, it
+# --- depends on the driver having certain code constructs. If the
+# --- driver changes significantly, the following sed edit may fail,
+# --- and require updating.
+# --- In particular, we have taken care that the sed replacement with
+# --- the keyword *mode* below only replaces the first instance in the
+# --- file. Since this keyword is frequently shared among I/O APIs,
+# --- it is possible that new code introduced into the target file,
+# --- nozzle_timing.py, may cause this sed edit to edit the wrong call.
+#
+# --- Edit the nozzle driver for:
+# ---- 20 steps
+# ---- no i/o
+# ---- desired file namings
+sed -e 's/\(nviz = \).*/\11000/g' \
+    -e 's/\(nrestart = \).*/\11000/g' \
+    -e 's/\(current_dt = \).*/\15e-8/g' \
+    -e 's/\(t_final = \).*/\11e-6/g' \
+    -e 's/y0_euler/nozzle-timing/g' \
+    -e 's/y0euler/nozzle-timing/g' \
+    -e 's/mode="wu"/mode="wo"/' \
+    -e 's/\(casename = \).*/\1"nozzle-timing"/g' < ./nozzle.py > ./nozzle_timing.py
```

[semi-important] How do you feel about this vs using `patch`? That'll at least fail noisily if it fails to apply, remedying some of the brittleness described in the comment.

---

In this case we have a driver code that is frequently changing, significantly. Not the variable names so much, but their values, their location, the comments around them, etc.

I like sed better in this case - precisely because it just keeps on trucking under arbitrary changes to the driver, but still replaces the things I want to replace.

Patch fails frequently because the driver is not in steady state - it frequently changes significantly enough to mess up the patching, and exactly in the places where we need to change it.

---

OK, fair enough. Fine by me. We'll know what we ran by the driver MD5.